### PR TITLE
Improving performance in the loops

### PR DESCRIPTION
**Pull request by BigLes**
forEach loop is 30% slover than regular for loop, that's why I purpose using regular for loops instead of forEach
https://jsperf.com/foreach-vs-loop/43

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,6 +6,6 @@
   "scripts": {
     "test": "echo \"Error: no test specified\" && exit 1"
   },
-  "version": "0.1.7",
+  "version": "0.1.8",
```
Comment (nmccready) [inline package.json]: You should not be editing this **ever**. This should be handled by `npm version` itself.

Comment (BigLes) [reply package.json]: reverted back to original

Comment (nmccready) [reply package.json]: 👍

**Comment by nmccready** [comment]
LGTM, this is probably relevant when we're trying to squeeze out React performance and make the argument that using global hooks are better than contexts.

**Comment by nmccready** [comment]
Note #11 PR does this as well. Yes this is more atomic, but why not I thought I'd give it a shot.

File: index.js [hunk 2]
```diff
@@ -1,8 +1,9 @@
 function setState(newState) {
+  const listenersLength = this.listeners.length;
   this.state = { ...this.state, ...newState };
-  this.listeners.forEach((listener) => {
-    listener(this.state);
-  });
+  for (let i = 0; i < listenersLength; i++) {
+    this.listeners[i](this.state);
+  }
```
Comment (nmccready) [inline index.js:6]: There is a filter iterator below this. This should probably be changed to a `for` loop as well.

Comment (BigLes) [reply index.js:6]: Changed as well

Comment (markmssd) [reply index.js]: While we’re at it, I would propose ++i instead of i++

**Comment by andregardi** [comment]
@BigLes 
Thank you for this.
I am sorry for get so long to reply.

Thank you also @nmccready 
I will be soon reviewing yours PRs too.

**Pull request merged**